### PR TITLE
Adds project summary grid front-end

### PR DESCRIPTION
**Pull request by illusional**
Partial changes from https://github.com/populationgenomics/sample-metadata/pull/117.

Adds a project summary table that allows you to select a project, and displays a paged list of samples in a table.

Technical:

- Moved some files (Documentation.tsx) for better project structure.
- Adds project-summary route for viewing a PAGED summary grid of a project
- Adds ability to store global context in react app
- Adds reactstrap UI library

### CODE REVIEW

File: web/src/project/ProjectSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    const directToken = searchParams.get('token')
+
+    const [summary, setSummary] = React.useState<ProjectSummaryResponse | undefined>()
+    const [suppliedDirectToken, setSuppliedDirectToken] = React.useState<boolean | undefined>(undefined)
+    const [pageTokens, setPageTokens] = React.useState<number[]>([])
+    const [isLoading, setIsLoading] = React.useState<boolean>(false)
+    const [error, setError] = React.useState<string | undefined>()
+    const [pageLimit, _setPageLimit] = React.useState<number>(PAGE_SIZES[0])
+
+    async function getProjectSummary(token: any, addToPageToken: boolean) {
```
Comment (daniaki) [inline web/src/project/ProjectSummary.tsx:32]: If you define a function within a component that uses state/context variables from the outer scope, the closure could become stale when React runs a state update and you might end up with strange bugs. For this particular use case `useCallback` is a hook provided by the `react` package such that your function will be re-defined if any dependencies change - in this case `[setIsLoading, globalContext, pageLimit, pageTokens, setPageTokens]`.  I'm not sure if there will be any bugs in this particular situation, but just a heads-up!

File: web/src/project/ProjectSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                }
+                setIsLoading(false)
+                setSummary(resp.data)
+            }).catch(er => {
+                setIsLoading(false)
+                setError(er.message)
+            })
+    }
+
+    function setPageLimit(e: React.ChangeEvent<HTMLInputElement>) {
```
Comment (daniaki) [inline web/src/project/ProjectSummary.tsx:50]: Same comment as above!

File: web/src/project/ProjectSummary.tsx [hunk 3]
```diff
@@ -59,15 +60,15 @@ export const ProjectSummary = () => {
...
 
-    // eslint:
     React.useEffect(() => {
         if (_.isUndefined(suppliedDirectToken)) {
             setSuppliedDirectToken(!!directToken)
         }
         getProjectSummary(directToken, false)
 
+        // retrigger if project changes, or pageLimit changes
     }, [globalContext.project, pageLimit])
```
[the remark this answers is not in the document]
Comment (illusional) [reply web/src/project/ProjectSummary.tsx:72]: I don't want this to re-run on those changes, so I've written this as a combination of useEffect and useCallback, where useCallback will update the function when any relevant functions / variables are updated, where I can still only cause the update to project summary on project / pageLimit changes.

[the remark this answers is not in the document]
Comment (daniaki) [reply web/src/project/ProjectSummary.tsx]: I see! You're original solution should work fine then, but this is great too.

**Review by daniaki — APPROVED** [review]

**Pull request merged**